extensions/0znzw/0tils.js

```diff
@@ -0,0 +1,1565 @@
+        if (args.encoding == "base64_old") {
+          return btoa(val);
+        } else if (args.encoding == "unicode") {
+          return val.charCodeAt(0);
+        } else if (args.encoding == "binary") {
+          return Array.from(val)
+            .map((each) => each.charCodeAt(0)
+              .toString(2))
+            .join(" ");
+        }
+      } else if (args.mode == "Decode") {
+        if (args.encoding == "base64_old") {
+          return atob(val);
+        } else if (args.encoding == "unicode") {
+          return String.fromCharCode(val);
+        } else if (args.encoding == "binary") {
+          return val.split(" ")
+            .map(
+              (x) => x = String.fromCharCode(parseInt(x, 2)))
+            .join("");
+        }
+      }
+    }
+    deleteSpriteNoConfirm(args) {
+      //TAKEN FROM SHOVEL UTILS, 
+      const target = vm.runtime.getSpriteTargetByName(args.SPRITE);
+      if (!target || target.isStage) {
+        return;
+      }
+      vm.deleteSprite(target.id);
+    }
+    deleteCostume(args) {
+      const target = vm.runtime.getSpriteTargetByName(args.SPRITE);
+      if (!target || target.isStage) {
+        return;
+      }
+      target.deleteCostume(target.getCostumeIndexByName(args.COSNAME));
+    }
+    //CUSTOM BROADCASTS
+    broadcast({
+      EVENT
+    }, util) {
+      if (EVENT == "onetime") {
+        if (this.ranOnce == undefined) {
+          util.startHats('SGCUtils_whenReceived', {
+            EVENT_OPTION: "onetime"
+          });
+          this.ranOnce = true;
+        }
+      } else {
+        util.startHats('SGCUtils_whenReceived', {
+          EVENT_OPTION: EVENT
+        });
+      }
+    }
+    //RESET ONETIME
+    resetOnetime() {
+      this.ranOnce = undefined;
+    }
+    //FILES
+    download(args) {
+      rawDownload(args.text, args.file);
+    }
+    newZip(args) {
+      var zip = new JSZip(); // eslint-disable-line
+      if (zips.includes(args.name)) return;
+      zips.push(args.name);
+      zips.push(zip);
+    }
+    zipNewFile(args) {
+      if (!zips.includes(args.name)) return;
+      const zipIndex = zips.indexOf(args.name) + 1;
+      const zip = zips[zipIndex];
+      zip.file(args.filename, args.filedata);
+      zips[zipIndex] = zip;
+    }
+    zipFinish(args) {
+      if (!zips.includes(args.name)) return;
+      const zipIndex = zips.indexOf(args.name) + 1;
+      const zip = zips[zipIndex];
+      zips.pop(zipIndex);
+      zips.pop(zipIndex - 1);
+      return zip.generateAsync({
+          type: "blob"
+        })
+        .then(function(content) {
+          return content;
+        });
+    }
+    //REDIRECTION
+    async linkopen(args) {// eslint-disable-line
+      //YEAH IK I AM SUPPOSED TO USE "await Scratch.redirect(url);" but it did not do what I expected.
+      if (args.target == "_self") {
+        window.location.href = args.url;// eslint-disable-line
+      }
+      window.open(args.url, args.target);// eslint-disable-line
```

Comment:
do not disable eslint. those warnings exist for a reason.

Reply:
eslint was disabled because it gave me issues with having to use the built in redirect which does not support _top, _parent, and framename(not used for obvious reasons)

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#target